### PR TITLE
docs: config chapter

### PR DESCRIPTION
**Pull request by onbjerg**
Adds a reference on how to configure reth using `reth.toml`, as well as describing some of the tradeoffs of increasing/decreasing values.

### CODE REVIEW

**Comment by onbjerg** [comment]
cc @mattsse

- when are peers banned? in reference to `[peers.reputation_weights]`
- what is `peers.refill_slots_interval`
- when do we use `low`/`medium`/`high`/`max` for peer backoffs?
- what is the diff between the session command and event buffer? command are from us, events are from them?

**Review by shekhirin — COMMENTED** [review]
good stuff

File: book/run/config.md [hunk 1]
```diff
@@ -1,2 +1,331 @@
...
+# The amount of accounts to process before writing the results to disk.
+#
+# Lower thresholds correspond to more frequent disk I/O (writes),
+# but lowers memory usage
+commit_threshold = 100000
+```
+
+### `storage_hashing`
+
+The account hashing stage builds a secondary table of account storages, where the key is the hash of the address and the slot, instead of the raw address and slot.
```
Comment (shekhirin) [inline book/run/config.md:170]: invalid description

**Review by shekhirin — APPROVED** [review]
LGTM

**Review by mattsse — APPROVED** [review]

**Pull request merged**